### DISCUSSION TITLE
Why properties with default value should be marked as optional?

### DISCUSSION THREAD

**Discussion by wong2** [typegoose/typegoose#492] [Q&A] [answered]
example from the [doc](https://typegoose.github.io/typegoose/docs/api/decorators/prop/#default): 

```ts
class Defaulted {
  @prop({ default: 'hello world' })
  public upperCase?: string; // mark as optional, because it will be defaulted
}
```

I don't understand why the property is marked as optional, since it will always have a default value, it should never be `undefined`.

**Comment by hasezoey** [comment, marked as answer]
this is for typescript, otherwise you will be *required* to input this property in something like `new Model({ here })` or `Model.create({ here })`
but this error will only come if tsconfig option `strict` is activated (so for both cases (option enabled or not), the signs `!` and `?` are included in the tests / documentation)

**Reply by wong2** [reply]
I have `strict: true` in tsconfig, but `Model.create({})` is not checking for types, even the _required_ props

**Reply by hasezoey** [reply]
do you have the latest `@types/mongoose` installed and not overwriting the `create` types (with the generics)?

**Reply by AasishP** [reply]
type checking is not working even when strict : true and latest @types/mongoose installed

**Reply by hasezoey** [reply]
please explain what "strictness" you need, maybe also read [FAQ: Why does new Model({}) not have types?](https://typegoose.github.io/typegoose/docs/guides/faq#why-does-new-model-not-have-types) and note that `@types/mongoose` is not required anymore since mongoose ~5.10

TL;DR: if i remember correctly, the `@types/mongoose` type were stricter about those types but in mongoose's own types they are merely a *suggestion*

**Reply by AasishP** [reply]
I want type cheacking to be enabled. How can it be done? and what would be the cons of doing so?

**Reply by hasezoey** [reply]
> I want type cheacking to be enabled. How can it be done? and what would be the cons of doing so?

most of the types are just *suggestions* because they are written in a way to not be enforced (because mongoose allows may ways to add properties in various stages), typegoose does not change that fact and unless you re-write those types yourself to be enforced, you will not get enforced types
